pkg/pillar/cmd/agent.go

```diff
@@ -0,0 +1,63 @@
+package cmd
+
+import (
+	"flag"
+	"github.com/lf-edge/eve/pkg/pillar/agentlog"
+	"github.com/lf-edge/eve/pkg/pillar/pidfile"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// AgentBase - Interface for all agents to use the AgentBase.
+// Each agent is expected to have a Context that implements
+// this interface.
+type AgentBase interface {
+	AgentBaseContext() *AgentBaseContext
+	AddAgentSpecificCLIFlags()
+	ProcessAgentSpecificCLIFlags()
+}
+
+// AgentBaseCLIParams - stores all the common cli params
+type AgentBaseCLIParams struct {
+	DebugOverride bool
+}
+
+// AgentBaseContext - a struct that represents a general agent context
+type AgentBaseContext struct {
+	CLIParams   AgentBaseCLIParams
+	ErrorTime   time.Duration
+	WarningTime time.Duration
+	AgentName   string
+}
+
+// processCLIFlags - Add flags common to all agents
+func processCLIFlags(agentBase AgentBase) {
+	debugPtr := flag.Bool("d", false, "Debug flag")
+	agentBase.AddAgentSpecificCLIFlags()
+	flag.Parse()
+	ctx := agentBase.AgentBaseContext()
+	ctx.CLIParams.DebugOverride = *debugPtr
+	agentBase.ProcessAgentSpecificCLIFlags()
+}
+
+// DoAgentStart - a general run function that will handle all of the common code for agents
+func DoAgentStart(agentSpecificContext AgentBase) error {
```

Comment:
I don't think this is the right name..

Eventually - the agent "Run" function will look like:

func Run(<pubsub driver>) {
   <create Agent Context>
   ctx.AgentBaseContext.Run()
}

It will never return. The inifinite loop will be in AgentBase. All other agent specific activity will happen only through the methods of AgentContext.

Lets name this method as Run()

Reply:
That's what it is was originally! I'll change it back :)